_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -751,7 +883,43 @@ mfxStatus VideoDECODEVP9_HW::DecodeFrameCheck(mfxBitstream *bs, mfxFrameSurface1
     if (NeedToReturnCriticalStatus(bs))
         return ReturningCriticalStatus();
 
-    MFX_CHECK_NULL_PTR2(surface_work, surface_out);
+    if (bs && !bs->DataLength)
+        return MFX_ERR_MORE_DATA;
+
+    MFX_CHECK_NULL_PTR1(surface_out);
+
+    *surface_out = nullptr;
+
+    if (bs == nullptr)
+    {
+        // clean DPB
+        for (mfxI32 ref_index = 0; ref_index < NUM_REF_FRAMES; ++ref_index)
+        {
+            if (m_frameInfo.ref_frame_map[ref_index] >= 0)
+            {
+                MFX_CHECK((m_FrameAllocator->DecreaseReference(m_frameInfo.ref_frame_map[ref_index]) == UMC::UMC_OK), MFX_ERR_UNKNOWN);
+                m_frameInfo.ref_frame_map[ref_index] = -1;
+            }
+        }
+        bool isDecodingFinished = false;
+        try
+        {
+            m_framesStorage->CompleteFrames();
```

Comment:
Method CompleteFrames should be the first statement in DecodeCheckFrame because it releases decoded frames and its reference.